samples/snippets/csharp/VS_Snippets_CLR/IO.Enumeration.FileSystemEnumerable/CS/Sample1.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.IO;
+using System.IO.Enumeration;
+
+namespace MyNamespace
+{
+    public class MyClassCS
+    {
+        static void Main()
+        {
+            /*
+                Create the following file tree:
+                    C:\Users\username\AppData\Local\Temp\testfolder\file1.txt
+                    C:\Users\username\AppData\Local\Temp\testfolder\file2.mp3
+                    C:\Users\username\AppData\Local\Temp\testfolder\testsubfolder\file3.txt
+            */
+
+            string pathDir = Path.Combine(Path.GetTempPath(), "testfolder");
+            string pathFile1 = Path.Combine(pathDir, "file1.txt");
+            string pathFile2 = Path.Combine(pathDir, "file2.mp3");
+
+            Directory.CreateDirectory(pathDir);
+            File.CreateText(pathFile1).Dispose();
+            File.CreateText(pathFile2).Dispose();
+
+            string pathSubdir = Path.Combine(pathDir, "testsubfolder");
+            string pathFile3 = Path.Combine(pathSubdir, "FILE3.TXT");
+
+            Directory.CreateDirectory(pathSubdir);
+            File.CreateText(pathFile3).Dispose();
+
+            // Create the enumeration object
+
+            var options = new EnumerationOptions()
+            {
+                RecurseSubdirectories = true
+            };
+
+            static string myTransform(ref FileSystemEntry entry) =>
+               entry.ToFullPath();
+
+            static bool myPredicate(ref FileSystemEntry entry) =>
+               !entry.IsDirectory &&
+               Path.GetExtension(entry.ToFullPath()).Equals(".txt", StringComparison.InvariantCultureIgnoreCase);
```

Comment:
I tried to write the same example in Visual Basic but the predicate and the transform were giving me headaches. I know there are no local methods in VB, but I couldn't figure out the syntax for the delegate with a ByRef argument. I'll give it another try soon.

Also, I was getting a build failure when trying to use `ByRef FileSystemEntry`:

FileSystemEntry' is obsolete: 'Types with embedded references are not supported in this version of your compiler.

I'll ask someone in my team why this is happening.